pkg/test/testdata/testdata/otel/otlp-agent-forwarder.tmpl

```diff
@@ -0,0 +1,42 @@
+receivers:
+  prometheus:
+    config:
+      scrape_configs:
+        - job_name: 'otel-collector'
+          scrape_interval: 1s
+          static_configs:
+            - targets: [':{{ .TelemetryPort }}']
+processors:
+  resourcedetection/system:
+    detectors: ["system"]
+    system:
+      hostname_sources: ["lookup"]
+exporters:
+  {{ if .OTLPExporterEndpoint }}
+  otlp:
```

Comment:
Locally, I couldn't get `forwarder_test.go` to work: 
```
  2023-09-29T22:29:14.078Z	error	exporterhelper/queued_retry.go:401	Exporting failed. The error is not retryable. Dropping data.	{"kind": "exporter", "data_type": "metrics", "name": "otlp", "error": "Permanent error: rpc error: code = Unimplemented desc = grpc: Decompressor is not installed for grpc-encoding \"gzip\"", "dropped_items": 16}
```

The fix was to append `compression: none` to the OTLP exporter.
```suggestion
  otlp:
    compression: none    
```